### PR TITLE
Bonnie 1580 result datetime milliseconds fix

### PR DESCRIPTION
**Pull request by c-monkey**
- [ ] After deployment, update fixture with rematerialized test cases so they pass. Doesn't affect the unit tests, but it would be nice for test cases to pass.

Fixes a unit conversion bug where DateTime results in data criteria and in components are represented as seconds rather than milliseconds for CQL times.

Pull requests into Bonnie require the following. Submitter and reviewer should :white_check_mark: when done. For items that are not-applicable, note it's not-applicable ("N/A") and :white_check_mark:.

**Submitter:**
- [X] This pull request describes why these changes were made.
- [X] This PR is into the correct branch.
- [X] JIRA ticket for this PR: https://jira.mitre.org/browse/BONNIE-1580
- [x] JIRA ticket links to this PR
- [X] Code diff has been done and been reviewed (it **does not** contain: additional white space, not applicable code changes, debug statements, etc.)
- [X] If UI changes have been made, google WAVE plug-in has been executed to ensure no 508 issues were introduced. N/A
- [X] Tests are included and test edge cases
- [X] Tests have been run locally and pass (remember to update Gemfile when applicable)
- [X] Test fixtures updated and documented as necessary ( see [internal wiki](https://gitlab.mitre.org/bonnie/internal-documentation/wikis/testing#test-fixtures) )
- [X] Code coverage has not gone down and all code touched or added is covered. 
     * In rare situations, this may not be possible or applicable to a PR. In those situations:
         1. Note why this could not be done or is not applicable here: 
         2. Add TODOs in the code noting that it requires a test
         3. Add a JIRA task to add the test and link it here: 

Branch | Back End Coverage | Front End Coverage
-- | -- | --
master | 1763 / 1972 LOC (89.4%) covered | Statements   : 59.45% ( 7283/12251 )<br>Branches     : 43.46% ( 1982/4561 )<br>Functions    : 49.81% ( 1207/2423 )<br>Lines        : 57.32% ( 6539/11408 )
(pr branch) | 1769 / 1972 LOC (89.71%) covered | Statements   : 59.45% ( 7283/12251 )<br>Branches     : 43.46% ( 1982/4561 )<br>Functions    : 49.81% ( 1207/2423 )<br>Lines        : 57.32% ( 6539/11408 )

- [x] Automated regression test(s) pass - tests do pass with no changes. Unfortunately, the real test here would be a rebuildPatients rake task that recalculates all of the extracted fields out of the source_data_criteria.

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA test links: https://jira.mitre.org/browse/BONNIE-1635
- [x] Justification for using JIRA tests: verification of fix to external ticket. All new functionality is covered by the unit tests, this is just for sake of completeness. 
- [x] JIRA tests have been added to sprint

**Reviewer 1:**

Name: @mayerm94 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

**Reviewer 2:**

Name: @edeyoung 
- [x] Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose
- [x] The tests appropriately test the new code, including edge cases
- [x] You have tried to break the code

If JIRA tests were used to supplement or replace automated tests:
- [x] JIRA tests have been run and pass
- [x] You agree with the justification for use of JIRA tests or have provided input on why you disagree

### CODE REVIEW

**Comment by c-monkey** [comment]
Note, this fixes https://oncprojectracking.healthit.gov/support/browse/BONNIE-485. To test you need to modify the Assessment RESULT value to be a dateTime.

File: lib/measures/patient_builder.rb [hunk 1]
```diff
@@ -184,7 +184,7 @@ def self.derive_values(entry, values, value_sets)
           derived << CodedResultValue.new({codes:codes, description: vs["display_name"]})
         elsif result_value['type'] == 'TS'
           # Recycling the use of Range/PhysicalQuantity for TimeStamps
-          time = HQMF::Range.from_json('low' => {'value' => result_value['value'], 'unit' => 'UnixTime'})
+          time = HQMF::Range.from_json('low' => {'value' => result_value['value']/1000, 'unit' => 'UnixTime'})
```
Comment (edeyoung) [inline lib/measures/patient_builder.rb]: add comments explaining this change

File: test/functional/patient_builder_test.rb [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+
+    record = Record.where(last: 'Numerator').first
+    # clear calculated value from fixture
+    record.assessments[0]['values'][0]['scalar'] = '0'
+    Measures::PatientBuilder.rebuild_patient(record)
+    assert_equal('UnixTime', record.assessments[0]['values'][0]['units'])
+    # this new calculated value should be in seconds
+    assert_equal(1341129600, record.assessments[0]['values'][0]['scalar'])
+
+    record = Record.where(last: 'NoNumerator').first
```
Comment (edeyoung) [inline test/functional/patient_builder_test.rb]: why are two records tested? does numerator vs no numerator change things? if so, recommend leaving a comment.

**Review by edeyoung — CHANGES_REQUESTED** [review]
see comments

File: test/functional/patient_builder_test.rb (hunk 2 above)
Comment (c-monkey) [reply test/functional/patient_builder_test.rb]: This only changes things calculated in the front-end. I removed the redundant test cases. The JIRA test demonstrates how the newly converted times work correctly with the logic.

**Review by edeyoung — APPROVED** [review]

**Review by mayerm94 — CHANGES_REQUESTED** [review]
I think there may be a problem with serializing the result datetime out of the database. When I open CMS879, the patient 'Numerator Test' is not a part of the IPP, if I delete the Result DateTime and add it back in (same date), the patient then passes. NOTE: I can only replicate this with a fresh DB snapshot, once I modify the result date, save and re-open it works fine.

**Comment by c-monkey** [comment]
This is because the fixtures are created on the bonnie.healthit.gov website which doesn't have the new code.

I guess after deployed we should update the results and replace the fixture. To update, just, change the patient's name and change focus, then change it back. Whatever is required to do the materialize.

**Review by mayerm94 — APPROVED** [review]

**Pull request merged**